README.md

```diff
@@ -1,3 +1,448 @@
+        int? completedByPersonId,
+        CancellationToken cancellationToken) // Can use CancellationToken directly (equivalent to the RequestAborted property)
+    {
+        IQueryable<Todo> query = _repository.Todos;
+        
+        if (id.HasValue)
+            query = query.Where(x => x.Id == id);
+        if (ids != null)
+            query = query.Where(x => ids.Contains(x.Id));
+        if (completedByPersonId != null)
+            query = query.Where(x => x.CompletedByPersonId == completedByPersonId);
+
+        return await query.ToListAsync(cancellationToken);
+    }
+
+    // Single item query
+    public async Task<Todo> TodoAsync(int id, CancellationToken cancellationToken)
+    {
+        return await _repository.Todos
+            .Where(x => x.Id == id)
+            .SingleOrDefaultAsync(cancellationToken);
+    }
+}
+```
+
+### Root Mutation Type
+
+Mutations are easily implemented using `DIObjectGraphBase` (which is the same
+as `DIObjectGraphBase<object>`, useful when the object itself is unimportant):
+
+```csharp
+public class Mutation : DIObjectGraphBase
+{
+    private readonly IRepository _repository;
+
+    public Mutation(IRepository repository)
+    {
+        _repository = repository;
+    }
+
+    public async Task<Todo> AddTodoAsync(string title, string notes)
+    {
+        // Use RequestAborted for cancellation support
+        var todo = new Todo {
+            Title = title,
+            Notes = notes,
+        };
+        return await _repository.AddTodoAsync(todo, RequestAborted);
+    }
+
+    public async Task<Todo?> SetCompleteAsync(int id, int completedByPersonId)
+    {
+        var todo = await _repository.GetTodoAsync(id, RequestAborted);
+        if (todo == null)
+            return null;
+            
+        if (todo.Completed)
+            throw new ExecutionError($"Task id {id} has already been completed");
+
+        todo.Completed = true;
+        todo.CompletedByPersonId = completedByPersonId;
+        todo.CompletionDate = DateTime.Now;
+        
+        await _repository.SaveChangesAsync(RequestAborted);
+        return todo;
+    }
+}
+```
+
+## Important Notes
+
+- Classes inheriting from `DIObjectGraphBase` **must** be registered with a Transient lifetime in the DI container if they implement `IDisposable`
+- The `Source` property provides type-safe access to the parent object being resolved
+- Static methods in resolver classes will not create an instance of the class, useful for simple property resolution
+- The library fully supports GraphQL.NET's data loader pattern for efficient batching and caching of data fetching operations
+- Use the `RequestAborted` property or `CancellationToken` parameter for cancellation support
+- Throw `ExecutionError` to return specific error messages to GraphQL clients
+- Can be used alongside traditional GraphQL.NET types, allowing for gradual adoption
+- All standard GraphQL.NET context properties are available through the `Context` property
```

Comment:
i really think we should add examples here, like coding examples with code blocks